<a href="https://colab.research.google.com/github/shash365/iitkgp-aiml/blob/main/Linear_Regression.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
# import csv file into colab

from google.colab import files
uploaded = files.upload()

Saving LR_data.csv to LR_data.csv


In [2]:
# import libraries

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:
# Fetch data
data = pd.read_csv('LR_data.csv')
data.head()

Car       Model  Volume  Weight  CO2
0      Toyoty        Aygo    1000     790   99
1  Mitsubishi  Space Star    1200    1160   95
2       Skoda      Citigo    1000     929   95
3        Fiat         500     900     865   90
4        Mini      Cooper    1500    1140  105

In [4]:
data.shape

(36, 5)

In [5]:
# Selecting features by analyzing the data. Here, in order to predict CO2 emission, car and model name are irrelevant.
# So Volumne and Wight will matter

In [6]:
X = data[['Volume','Weight']]
y = data['CO2']

In [9]:
# Import Lienar model

from sklearn.linear_model import LinearRegression

In [10]:
lr = LinearRegression()

In [11]:
lr.fit(X, y)

LinearRegression()

In [13]:
# Lets predict CO2 emission when Volume and Weight car given for a car

co2_pred = lr.predict([[2500,1800]])
print(f'CO2 emission for a car when Volume :2500 and Weight: 1800 - {co2_pred}')

CO2 emission for a car when Volume :2500 and Weight: 1800 - [112.7995682]


/usr/local/lib/python3.11/dist-packages/sklearn/utils/validation.py:2739: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


In [15]:
# Lets take input from user and predict the CO2 emission for a car

Vol = int(input('Enter Volume of Car'))
Weg = int(input('Enter Weight of Car'))
co2_pred = lr.predict([[Vol,Weg]])
print(f'CO2 emission for the car when Volume :{Vol} and Weight: {Weg} - {co2_pred}')

Enter Volume of Car2200
Enter Weight of Car1200
CO2 emission for the car when Volume :2200 and Weight: 1200 - [105.92742258]


/usr/local/lib/python3.11/dist-packages/sklearn/utils/validation.py:2739: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


Finally, we predicted that a car with 2.2-liter engine, and a weight of 1200 kg, will release approximately 105.93 grams of CO2 for per kilometer it drives.

In [17]:
# lets calculate the coefficients and intercept
lr.coef_

array([0.00780526, 0.00755095])

In [18]:
lr.intercept_

np.float64(79.69471929115939)